runtime/onert/core/src/backend/cpu_common/DynamicTensorManager.cc

```diff
@@ -73,7 +73,7 @@ void DynamicTensorManager::applyShape(const ir::OperandIndex &ind, const ir::Sha
 
       setShape(tensor.get(), new_shape);
       tensor->set_dynamic();
-      allocTensorMem();
+      allocTensorMem(true);
```

Comment:
I don't clearly understand this code.
after deallocating, I think tensor will have nullptr alloc. 
However, here, it looks that the tensor still have some alloc.
Could you explain more?
IMHO, If you add your explanation in the code as comment, it would better since people like me who read code will be curious in future.

Reply:
I haven't seen deallocating dynamic tensor properly.
I will loot at it in detail. Thanks for your pointing out.

Reply:
@hyunsik-yoon 
`_dynamic_mem_mgr->deallocate(ind);` does not make `_allocator` of `Tensor` to nullptr. It only release `_base` of `Allocator`
I have two other suggestions.
```patch
-      _dynamic_mem_mgr->deallocate(ind);
+      // Release memory from tensor
+      tensor->overwriteBuffer(nullptr);
 
       setShape(tensor.get(), new_shape);
       tensor->set_dynamic();
       allocTensorMem();
```
```patch
-      _dynamic_mem_mgr->deallocate(ind);
- 
       setShape(tensor.get(), new_shape);
       tensor->set_dynamic();
-      allocTensorMem();
+      allocTensorMem(true);
```

Reply:
`allocTensorMem(true); `
Oh. right. I see. Thanks.

for `-      _dynamic_mem_mgr->deallocate(ind);` @ragmani will mention.

Reply:
`_dynamic_mem_mgr->deallocate(ind);` should be exists so that `std::shared_ptr<Allocator>`'s `use_count` decreases.